### PR TITLE
EW-525: Add GET /personenkontexte endpoint

### PR DESCRIPTION
**Pull request by mkreuzkam-cap**
# Description
- Add endpoint GET /api/personenkontexte to get all Personenkontexte (supports filters and pagination)
- /api/person/{personId}/personenkontexte now also supports pagination
- Add PersonenkontextScope and use it in findAll for Personenkontexte
- Use SichtfreigabeType instead of boolean in personenkontext entity and DO (requires re-initializing the database)

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/EW-525

## Approval for review
- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

File: src/modules/person/api/personenkontext.controller.ts [hunk 1]
```diff
@@ -62,4 +68,40 @@ export class PersonenkontextController {
             throw error;
         }
     }
+
+    @Get()
+    @ApiOkResponse({
+        description: 'The personenkontexte were successfully returned.',
+        type: [PersonendatensatzResponse],
```
Comment (Fshmit) [inline src/modules/person/api/personenkontext.controller.ts:75]: headers: PagingHeadersObject is missing

File: src/modules/person/api/personenkontext-query.params.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import { AutoMap } from '@automapper/classes';
 import { ApiProperty } from '@nestjs/swagger';
 import { IsEnum, IsOptional, IsString } from 'class-validator';
-import { Personenstatus, Rolle } from '../domain/personenkontext.enums.js';
-import { SichtfreigabeType } from './personen-query.param.js';
+import { PagedQueryParams } from '../../../shared/paging/paged.query.params.js';
+import { Personenstatus, Rolle, SichtfreigabeType } from '../domain/personenkontext.enums.js';
 
-export class PersonenkontextQueryParams {
+export class PersonenkontextQueryParams extends PagedQueryParams {
```
Comment (Fshmit) [inline src/modules/person/api/personenkontext-query.params.ts:7]: Types in some AutoMap annotations are missing. see person-query.param.ts

File: src/modules/person/domain/personenkontext.service.ts [hunk 3]
```diff
@@ -30,10 +33,30 @@ export class PersonenkontextService {
...
     public async findAllPersonenkontexte(
         personenkontextDo: PersonenkontextDo<false>,
-    ): Promise<Result<PersonenkontextDo<true>[], DomainError>> {
-        const personenkontexte: PersonenkontextDo<true>[] = await this.personenkontextRepo.findAll(personenkontextDo);
+        offset?: number,
+        limit?: number,
+    ): Promise<Paged<PersonenkontextDo<true>>> {
+        const scope: PersonenkontextScope = new PersonenkontextScope()
+            .findBy({
+                personId: personenkontextDo.personId,
```
Comment (Fshmit) [inline src/modules/person/domain/personenkontext.service.ts:41]: is the personID defined at this position?

File: src/modules/person/api/personenkontext-query.params.ts (hunk 2 above)
Comment (mkreuzkam-cap) [reply src/modules/person/api/personenkontext-query.params.ts:7]: Done

File: src/modules/person/api/personenkontext.controller.ts (hunk 1 above)
Comment (mkreuzkam-cap) [reply src/modules/person/api/personenkontext.controller.ts:76]: Done

File: src/modules/person/api/personenkontext.controller.ts [hunk 4]
```diff
@@ -62,4 +69,41 @@ export class PersonenkontextController {
...
+            result.items,
+            PersonenkontextDto,
+            PersonenkontextdatensatzResponse,
+        );
+        const response: PagedResponse<PersonenkontextdatensatzResponse> = new PagedResponse({
+            items: responseItems,
+            total: result.total,
+            offset: result.offset,
+            limit: result.limit,
+        });
```
Comment (psachmann) [inline src/modules/person/api/personenkontext.controller.ts:105]: Between the consts we don't need blank lines.

File: src/modules/person/domain/personenkontext.service.spec.ts [hunk 5]
```diff
@@ -118,30 +119,35 @@ describe('PersonenkontextService', () => {
...
 
-                const result: Result<PersonenkontextDo<true>[], DomainError> =
-                    await personenkontextService.findAllPersonenkontexte(personenkontextDoWithQueryParam);
-                expect(result).toEqual<Result<PersonenkontextDo<true>[], DomainError>>({
-                    ok: true,
-                    value: personenkontexte,
+                const result: Paged<PersonenkontextDo<true>> = await personenkontextService.findAllPersonenkontexte(
+                    personenkontextDoWithQueryParam,
+                );
+                expect(result).toEqual<Paged<PersonenkontextDo<true>>>({
```
Comment (psachmann) [inline src/modules/person/domain/personenkontext.service.spec.ts:130]: Missing blank line.

File: src/modules/person/domain/personenkontext.service.spec.ts [hunk 6]
```diff
@@ -118,30 +119,35 @@ describe('PersonenkontextService', () => {
...
 
         describe('When no personenkontexte are found', () => {
             it('should return a result with an empty array', async () => {
                 const personenkontext: PersonenkontextDo<false> = DoFactory.createPersonenkontext(false);
-                personenkontextRepoMock.findAll.mockResolvedValue([]);
+                personenkontextRepoMock.findBy.mockResolvedValue([[], 0]);
                 mapperMock.map.mockReturnValue(personenkontext as unknown as Dictionary<unknown>);
-                const result: Result<PersonenkontextDo<true>[], DomainError> =
+                const result: Paged<PersonenkontextDo<true>> =
                     await personenkontextService.findAllPersonenkontexte(personenkontext);
```
Comment (psachmann) [inline src/modules/person/domain/personenkontext.service.spec.ts:145]: Missing blank line.

**Review by psachmann — CHANGES_REQUESTED** [review]

File: src/modules/person/api/find-personenkontext.dto.ts [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
-import { Rolle, Personenstatus } from '../domain/personenkontext.enums.js';
-import { SichtfreigabeType } from './personen-query.param.js';
+import { PagedDto } from '../../../shared/paging/paged.dto.js';
+import { Personenstatus, Rolle, SichtfreigabeType } from '../domain/personenkontext.enums.js';
 
-export class FindPersonenkontextDto {
+export class FindPersonenkontextDto extends PagedDto {
     @AutoMap()
-    public personId!: string;
+    public personId?: string;
```
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/api/find-personenkontext.dto.ts:7]: For the existing endpoint (/person/{id}/personenkontexte) this was always set, but the added one doesn't have a filter of that kind so it's empty. I thought I make this field optional to make that more obvious.

File: src/modules/person/api/person-id.response.ts
[the remark this answers is not in the document]
Comment (mkreuzkam-cap) [reply src/modules/person/api/person-id.response.ts:1]: It's part of a response. According to SchulConneX the response of this new endpoint is a Personendatensatz with a person which only has the id field set (v1.3, page 99). Is there a better name for this type of class?

File: src/modules/person/domain/personenkontext.service.spec.ts (hunk 5 above)
Comment (mkreuzkam-cap) [reply src/modules/person/domain/personenkontext.service.spec.ts]: Done

File: src/modules/person/api/personenkontext.controller.ts (hunk 4 above)
Comment (mkreuzkam-cap) [reply src/modules/person/api/personenkontext.controller.ts]: Done

File: src/modules/person/domain/personenkontext.service.spec.ts (hunk 6 above)
Comment (mkreuzkam-cap) [reply src/modules/person/domain/personenkontext.service.spec.ts:147]: Done

File: src/modules/person/api/person-id.response.ts
[the remark this answers is not in the document]
Comment (psachmann) [reply src/modules/person/api/person-id.response.ts:1]: I just asked, because it looked wired.

**Review by Fshmit — APPROVED** [review]

**Review by psachmann — APPROVED** [review]

**Pull request merged**